lib/ohai/mixin/train_helpers.rb

```diff
@@ -0,0 +1,45 @@
+#
+# Copyright:: Copyright (c) Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "chef-utils/dsl/train_helpers" unless defined?(ChefUtils::DSL::TrainHelpers)
+require_relative "chef_utils_wiring" unless defined?(Ohai::Mixin::ChefUtilsWiring)
+
+module Ohai
+  module Mixin
+    module TrainHelpers
+      include ChefUtils::DSL::TrainHelpers
+      include ChefUtilsWiring
+
+      def file_read(path)
+        file_open(path).readlines.join("\n")
```

Comment:
```suggestion
        file_open(path).read
```

Should be equivalent, unless we need strictly one newline between lines.

Reply:
yeah that one was bugging me a lot, to get it going i just copypasta'd what was in the other PR.

so the one complexity of this is that it could be either an IO from the filesystem or a StringIO coming roundabout from train, but ti seems like IO#read and StringIO#read are the same, and we definitely did NOT want that behavior between newlines, so i think that's also a bugfix.